lib/grpc/server/adapters/cowboy.ex

```diff
@@ -42,13 +44,24 @@ defmodule GRPC.Server.Adapters.Cowboy do
         {:ranch_tcp, :cowboy_clear}
       end
 
-    {ref, mfa, type, timeout, kind, modules} =
-      :ranch.child_spec(ref, transport, trans_opts, protocol, proto_opts)
+    case :ranch.child_spec(ref, transport, trans_opts, protocol, proto_opts) do
```

Comment:
Could you add a comment here and on the dialyzer warning explaining why these pieces of code are here?

Reply:
It's just a no_match/never match warning since `:ranch` version 1.8.0 will always return a tuple lilke current implementation, while new version (2.x.x) will always return map.
So, even though we update ranch the warning still exist because it'll just return one type of child_spec, a tuper or a map
And turned out we can't update `:ranch` because `:cowlib` still dependent on version 1.8.0

I think I'll close my PR, since it'd not be generic use cases for having support for both of supervisor child_spec